stdlib/InteractiveUtils/src/codeview.jl

```diff
@@ -278,17 +278,29 @@ Keyword argument `debuginfo` may be one of source (default) or none, to specify
 
 See also: [`@code_llvm`](@ref), [`code_warntype`](@ref), [`code_typed`](@ref), [`code_lowered`](@ref), [`code_native`](@ref).
 """
-function code_llvm(io::IO, @nospecialize(f), @nospecialize(types=Base.default_tt(f));
-                   raw::Bool=false, dump_module::Bool=false, optimize::Bool=true, debuginfo::Symbol=:default,
-                   params::CodegenParams=CodegenParams(debug_info_kind=Cint(0), debug_info_level=Cint(2), safepoint_on_entry=raw, gcstack_arg=raw))
-    d = _dump_function(f, types, false, false, raw, dump_module, :intel, optimize, debuginfo, false, params)
+function code_llvm end
+
+struct LLVMCode
+    str::String
+end
+
+function Base.show(io::IO, code::LLVMCode)
```

Comment:
```suggestion
function Base.show(io::IO, ::MIME"text/plain", code::LLVMCode)
```

To Jameson's point, for the two argument `show` the docstring does say the following.
> The representation used by show generally includes Julia-specific formatting and type information, and should be parseable Julia code when possible.

The main task is backwards compatible behavior for `display`ing the code, so I think the above should really be the pretty version for "text/plain".

```
julia> c = code_llvm(identity, Tuple{Int})
;  @ operators.jl:522 within `identity`
define i64 @julia_identity_396(i64 signext %0) #0 {
top:
  ret i64 %0
}


julia> string(c)
"LLVMCode(\";  @ operators.jl:522 within `identity`\\ndefine i64 @julia_identity_396(i64 signext %0) #0 {\\ntop:\\n  ret i64 %0\\n}\\n\")"
```

Reply:
Yeah, but that seems a complicated change for this PR to be breaking with the API for the other `code_*` function in this package (code_warntype)

Reply:
Most of the `code_*` functions currently return `nothing`. The breaking API change is that they now return something. Otherwise, the effect is largely the same: the displayed text is unchanged.

The new aspect is that a struct instance is returned and it can be stringified.

I'm unsure which part you are saying is complicated.